### PR TITLE
dts: silabs: Add SoC level compatible

### PR DESCRIPTION
**Pull request by galak**
Add compatible for all the SoC dtsi files.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by mnkp** [comment]
As I understand this PR is meant to prepare for generating part number, SoC series and SoC family information from DT? This looks good. The only unclear part is how to generate SoC family string. Taking following as an example
```
compatible = "silabs,efr32fg1p133f256gm48", "silabs,efm32fg1p", "silabs,efm32", "simple-bus";
```
the third property is defined as "silabs,efm32" while the directory structure we are using is 'silabs_exx32'. Are we planning to modify soc/ directory layout?

One more thing. Since we are introducing compatible property for the soc DT node shouldn't we provide also the binding. I mean the .yaml file?

**Comment by galak** [comment]
> the third property is defined as "silabs,efm32" while the directory structure we are using is 'silabs_exx32'. Are we planning to modify soc/ directory layout?

What do we need that for?  How do you plan on using it?

**Comment by galak** [comment]
Take a look at PR #12428

File: dts/arm/silabs/efr32fg1p133f256gm48.dtsi [hunk 1]
```diff
@@ -13,6 +13,8 @@
 	};
 
 	soc {
+		compatible = "silabs,efr32fg1p133f256gm48", "silabs,efm32fg1p", "silabs,efm32", "simple-bus";
```
Comment (chrta) [inline dts/arm/silabs/efr32fg1p133f256gm48.dtsi]: "silabs,efm32fg1p" should be "silabs,efr32fg1p"

File: dts/arm/silabs/efr32mg12p332f1024gl125.dtsi [hunk 2]
```diff
@@ -13,6 +13,8 @@
 	};
 
 	soc {
+		compatible = "silabs,efr32mg12p332f1024gl125", "silabs,efr32mg", "silabs,efr32", "simple-bus";
```
Comment (mnkp) [inline dts/arm/silabs/efr32mg12p332f1024gl125.dtsi]: "silabs,efr32mg" should be ""silabs,efr32mg12p""

**Review by mnkp — APPROVED** [review]

**Pull request merged**